scripts/perf/haswellx/hsx-metrics.json

```diff
@@ -156,31 +156,31 @@
     },
     {
         "BriefDescription": "Ratio of number of code read requests missing last level core cache (includes demand w/ prefetches) to the total number of completed instructions",
-        "MetricExpr": "(cbox@UNC_C_TOR_INSERTS.MISS_OPCODE\\,filter_opc\\=0x181@ + cbox@UNC_C_TOR_INSERTS.MISS_OPCODE\\,filter_opc\\=0x191@) / INST_RETIRED.ANY",
+        "MetricExpr": "llc_data_read_mpi_demand_plus_prefetch",
```

Comment:
This metric is written in terms of itself, ie it is recursive and would lock up if perf tried to evaluate it. There is a test in perf that detects this:

https://git.kernel.org/pub/scm/linux/kernel/git/perf/perf-tools-next.git/tree/tools/perf/util/metricgroup.c#n918

It may be worth copying the json here into a perf tree and running the tests to detect issues like this.

Reply:
@captain5050, this is metric ```llc_code_read_mpi_demand_plus_prefetch``` written in terms of  ```llc_data_read_mpi_demand_plus_prefetch```. They are two different metrics with very similar names. 

But I think this is also having the same issue like the SKX metrics. We will work on this one.

Reply:
Thanks, could we squash some (all maybe) of the changes? It is hard to review patches when there are fixes on the patches later in the series.

Reply:
@edwarddavidbaker, I guess I could make this PR to compare against the ```TMA-4.8-Release``` branch instead of the ```main``` branch to get it easier to read? What's your suggestion? Thanks!

Reply:
- Do we need to pull requests?
- Ian can comment with his preferences. I typically use interactive rebases (and force push the branch) to incorporate feedback / tweaks. This in essence creates a new patch set and is the typical work flow on gerrit. As a concrete example here, all of Caleb's commits would be squished into a single 4.8 change. Your changes then update the output directory `scripts/perf`, and can be squished into a single commit.

Reply:
With gerrit you can keep the magic Change-Id tag and gerrit time orders changes uploaded with it. I'm not sure how this works on github. My point of pain is trying to review many too large to open changes on github with fixes then layered on top in different patches. This makes it hard to say whether an issue is or isn't still present without mentally squashing the changes.

Reply:
One or two PRs, both are fine. But I think I don't have write access to Caleb's branch. I'm not sure how to push changes to that PR.

Reply:
@captain5050, now we only have changes to the script/perf/ directory remain in this PR. Does this work better?